catroid/src/main/java/org/catrobat/catroid/sensing/CollisionInformation.java

```diff
@@ -189,7 +197,7 @@ public void loadOrCreateCollisionPolygon() {
 			}
 
 			if (found) {
-				end = finalVertices.get(polygonNumber).get(finalVertices.get(polygonNumber).size() - 1);
+				end = currentPolygonVertex.get(currentPolygonVertex.size() - 1);
```

Comment:
how about you move this line of code to line 195 (before the `break`) and then get rid of the `boolean found`, for this loop. I know you still need it for the one below, so this might not be too much of an improvement, but hey, baby steps.